Unity-Technologies/ScriptableRenderPipeline
# Fixed resource reloader error

### Purpose of this PR

The first time the library is generated on an Universal project, the resource reloader would display an error about a shader file not found. This happened due to the way the Asset Database behaves and the fact that it calls OnEnable on assets before all the assets are loaded. In consequence, the first time the library is generated it would fail to find an asset in the database that wasn't loaded yet.

This fix is more of a "hack" as doing a proper fix would require a refactor of how the resource reloader works. Note that despite the reloading failing on the very first try, it will still work the second time as it's also called on the first domain reload that happens after the library generation.

---
### Testing status

Testing this is a bit painful and time consuming. The steps are as follow:

- Create an Universal project.
- Make sure a scene that makes use of Universal is loaded.
- Quit Unity.
- Delete the `Library` folder.
- Restart Unity.

Before this fix: you'd get an error in the console about a shader not being found. Resources would still be properly loaded.
After the fix: no more error and resources are still properly loaded.

---
### Overall Product Risks
**Technical Risk**: None

**Halo Effect**: None
